plugins/askrene/reserve.c

```diff
@@ -83,6 +83,41 @@ static bool remove(struct reserve *r, struct amount_msat amount)
 	return true;
 }
 
+/* Just like reserves_add, but used to rewind a failed reserves_remove. It
+ * cannot fail cause we don't want recursion. */
+static void reserves_add_no_fail(struct reserve_htable *reserved,
+				 const struct short_channel_id_dir *scidds,
+				 const struct amount_msat *amounts, size_t num)
+{
+	bool success;
+	for (size_t i = 0; i < num; i++) {
+		struct reserve *r = reserve_htable_get(reserved, &scidds[i]);
+		if (!r)
+			r = new_reserve(reserved, &scidds[i]);
+		assert(r);
```

Comment:
nit: you only need to assert in the case of a new reserve being created.

Reply:
Do you mean the assert should be within the scope of the `if` statement?